_posts/2019-09-09-kustomize.md

```diff
@@ -0,0 +1,185 @@
+Tiller is the central entry point and management component for all deployments using Helm 2.
+It keeps the state of all deployed manifests and groups them together as a single `release`.
+//TODO keep? The main disadvantage of Helm 2 is the lack of proper authorization / RBAC support as Tiller will run with almost root-level privileges on the cluster.
+//TODO keep? Another issue in the scope of this blog post is that it does more than just configuration management, it provides very basic release management.
+
+Helm 3 is currently still in beta, but the reworked version should resolve a lot of the issues with Helm 2.
+Like Helm 2 it's still a templating engine which also manages releases. 
+Unlike Helm 2, it doesn't use the Tiller component anymore to manage all the state connected to the release.
+More information about Helm can be found in other blogpost ([Links](#useful-links))
+
+Kustomize is a Kubernetes configuration management tool that is provided as part of the `kubectl` command, behind the `-k` flag.
+Kustomize allows a user to use standard Kubernetes manifests and overlay any changes that they want to make using an overlay manifest.
+Unlike Helm, Kustomize only provides the configuration management and doesn't manage any state about the manifests it adapts.
+Unlike Helm and OpenShift Templates, the main goal of Kustomize is to allow users to adapt their existing manifests in any thinkable way, instead of templating some parts of the manifest that can be edited.
+
+## What is Kustomize and how to use it
+
+As briefly discussed, Kustomize is a configuration management tool that has been embedded into `kubectl`.
+Originally it was a separate tool and some functionality is still only available in the Kustomize binary and not in Kubectl.
+The documentation of Kustomize is therefore available in two parts, the core docs and the Kubectl docs ([Links](#useful-links)).
+
+As discussed in the Kustomize projects [readme](https://github.com/kubernetes-sigs/kustomize/) a Kustomize manifest exists out of two main structures: a base manifest and overlays.
+
+Important to note, especially when considering the usage of this tool, is what it doesn't do:
+* It doesn't manage deployments
+* It doesn't package applications in deployable artifacts
+* It doesn't manage secrets securly
+
+### Base manifest
+
+A base manifest is, in essence, a set of bare-bones Kubernetes manifests.
+For the scope of this blog post, a single base manifest will contain all configuration to deploy a single service.
+//TODO keep? Kustomize doesn't require this, but it seems like a good fit.
+
+Let's assume the following set of Kubernetes manifests:
+```
+├── deployment.yaml
+├── ingress.yaml
+└── service.yaml
+```
+
+These manifests can deploy one of our apps to a cluster.
+It will create a deployment, a service exposing the app to the cluster and an ingress object that will allow connections from outside of the cluster to be setup.
```

Comment:
```suggestion
It will create a deployment, a service exposing the app to the cluster and an ingress object that will allow connections from outside of the cluster to be set up.
```